packages/@uppy/file-input/src/FileInput.tsx

```diff
@@ -1,14 +1,40 @@
-import { UIPlugin } from '@uppy/core'
+import { h, type ComponentChild } from 'preact'
+import type { CSSProperties } from 'preact/compat'
+
+import { UIPlugin, Uppy, type UIPluginOptions } from '@uppy/core'
 import toArray from '@uppy/utils/lib/toArray'
-import { h } from 'preact'
+import type { Body, Meta } from '@uppy/utils/lib/UppyFile'
 
+// eslint-disable-next-line @typescript-eslint/ban-ts-comment
+// @ts-ignore We don't want TS to generate types for the package.json
 import packageJson from '../package.json'
-import locale from './locale.js'
+import locale from './locale.ts'
+
+
+interface FileInputOptions extends UIPluginOptions {
+  target: HTMLElement | string | null
+  pretty?: boolean,
+  inputName?: string,
+}
+interface FileInputState {
+  audioReady: boolean
+  recordingLengthSeconds: number
+  hasAudio: boolean
+  cameraError: null
+  audioSources: MediaDeviceInfo[]
+  currentDeviceId?: null | string | MediaStreamTrack
+  isRecording: boolean
+  showAudioSourceDropdown: boolean
+  [id: string]: unknown
+}
 
-export default  class FileInput extends UIPlugin {
+export default  class FileInput<M extends Meta, B extends Body> extends
+  UIPlugin<FileInputOptions, M, B, FileInputState> {
```

Comment:
You should be using `DefinePluginOpts` and pass `defaultOptions` (needs to be extracted from the constructor)

```
this.opts = { ...defaultOptions, ...opts }
```
should also be removed and inlined into the `super` call.

Reply:
I copy-pasted from Audio.tsx. should it also be using DefinePluginOpts?

Reply:
Maybe audio has no default options? If it has, then it should use it too.

Reply:
that was it!